### ISSUE TITLE
Test case tests/kernel/test_mem_safe fails on ARM hardware

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#387]
**_Reported by Genaro Saucedo Tejada:_**

When running test case 'tests/kernel/test_mem_safe' on a real hardware the test seems to hang while console ouptus the expected messages but hundreds of times.

On FRDM board the following message can be seen at the end of the terminal output before it apparently crashes:
```
***** Debug monitor exception (not implemented) *****
Fatal fault in ISR ! Spinning...
```
(using open ocd 0.10)

Note, this doesn't reproduce on ARM qemus

(Imported from Jira ZEP-436)

**Comment by nashif** [comment]
**_by Inaky Perez-Gonzalez:_**

Sometimes it will spew a lot of output before dying, sometimes it doesn't seem to die (after a max of 200 seconds i've tried this).

kernel git revision:

81f61bb x86: make GDT setup optional

**Comment by nashif** [comment]
**_by Inaky Perez-Gonzalez:_**

* [kk.txt](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-387/kk.txt) (size: 418502)

**Comment by nashif** [comment]
**_by Benjamin Walsh:_**

Is this running the test multiple times without restarting the board or the OS ?

**Comment by nashif** [comment]
**_by Inaky Perez-Gonzalez:_**

A full power cycle is done, then the boards are flashed, then reset to run.

Three variations happen:
- the test runs to completion in less than 60s
- the test fails to complete (we tried all the way to 200s)
- the test fails with the ISR failure notification

**Comment by nashif** [comment]
**_by Inaky Perez-Gonzalez:_**

_Peter Mitsis_  can you take over this for Ben?

**Comment by nashif** [comment]
**_by Peter Mitsis:_**

Sure.

FYI, I've got a few items of higher priority on my plate at the moment so I am not sure exactly when I will get a chance to get digging into it.

**Comment by nashif** [comment]
**_by Inaky Perez-Gonzalez:_**

No probs, PM will bug you from time to time -- or if you want to aim it for a particular release, that way it can be set and just let it be scheduled appropriately.

**Comment by nashif** [comment]
**_by Benjamin Walsh:_**

Pretty sure this must be fixed by 0e6aebf

**Comment by nashif** [comment]
**_by Sharron LIU:_**

Verified passed @platform_whitelist.
